libkbfs/folder_update_prepper.go

```diff
@@ -0,0 +1,1223 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"fmt"
+
+	"github.com/keybase/client/go/logger"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/kbfs/kbfsblock"
+	"github.com/keybase/kbfs/tlf"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+)
+
+// folderUpdatePrepper is a helper struct for preparing blocks and MD
+// updates before they get synced to the backend servers.  It can be
+// used for a single update or for a batch of updates (e.g. conflict
+// resolution).
+type folderUpdatePrepper struct {
+	config       Config
```

Comment:
> It makes components more testable: you only have to implement test versions of the methods actually used (or pass in stock test implementations of each interface).

How does the `Config` interface not give this?  Isn't exactly the same amount of work to populate certain fields in a `ConfigLocal` as to do it in this new one, except it makes the test code harder to understand because the types change from test to test?

I can understand the explicit dependency argument, but to me that's only useful if we ever split things out into more packages, and it's over-engineering to do it now, and in my opinion makes the code harder to follow for no reason.